### PR TITLE
feat: [IDP-1030]: Generic api modification for calling from Http dsl layer

### PR DESCRIPTION
**Pull request by devesh-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52057)

### CODE REVIEW

**Review by vikyathharekal — COMMENTED** [review]
> Reviewed 17 of 32 files at r1, 1 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52057)* status: 32 file(s) to review. 0 of 2 approvals obtained. 

___
*[`idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParser.java` line 9 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncf_MAdEo8u5QTnjZmS:-Ncf_MAdEo8u5QTnjZmT:bqf7nfb) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParser.java#L9)):*
> ```Java
> package io.harness.idp.scorecard.datapoints.parser;
> 
> public interface DataSourceDataPointParser {
> ```

How about renaming it to `DataPointParserFactory`?



_Code quote:_
```Java
DataSourceDataPointParser
```
___
*[`idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParserFactory.java` line 24 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncf_6u9AGb5RUTnCp7c:-Ncf_6u9AGb5RUTnCp7d:b-tw0c60) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParserFactory.java#L24)):*
> ```Java
>   HarnessDataPointParserFactory harnessDataPointParserFactory;
> 
>   public DataSourceDataPointParser getDataSourceDataPointParser(String identifier) {
> ```

How about renaming it to `getDataPointParserFactory()`?



_Code quote:_
```Java
getDataSourceDataPointParser
```
___
*[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncfbyr04r337ur6_UUv:-Ncfbyr04r337ur6_UUw:b-86phg1) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java#L21)):*
> ```Java
> @UtilityClass
> public class ValueParserUtils {
>   public Map<String, Object> getMapForPipelineUrlsInfo(String ciPipelineUrl, String cdPipelineUrl) {
> ```

Map<String, String> would be fine I guess



_Code quote:_
```Java
Object
```
___
*[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java` line 31 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-NcfcKEx3uN8g2HFHTs-:-NcfcKEyC_ZwFQplxV_e:b-lm6jmk) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java#L31)):*
> ```Java
>     Map<String, Object> dataPointsInfo = new HashMap<>();
>     dataPointsInfo.put(ValueParserConstants.DATA_POINT_VALUE_KEY, dataPointValue);
>     dataPointsInfo.put(ValueParserConstants.PIPELINE_URLS_FOR_CHECK_FAILURE, urlInfo);
> ```

Can we keep this generic? Same keys can be used in Github case as well. How about `{"value": "", "errorMessage": ""}`? cc - @vigneswara-propelo

**Comment by devesh-harness** [comment]
> > Reviewed 17 of 32 files at r1, 1 of 3 files at r2, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/52057)_ status: 32 file(s) to review. 0 of 2 approvals obtained.
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParser.java` line 9 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncf_MAdEo8u5QTnjZmS:-Ncf_MAdEo8u5QTnjZmT:bqf7nfb) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParser.java#L9)):_
> 
> > ```java
> > package io.harness.idp.scorecard.datapoints.parser;
> > 
> > public interface DataSourceDataPointParser {
> > ```
> 
> How about renaming it to `DataPointParserFactory`?
> 
> _Code quote:_
> 
> ```java
> DataSourceDataPointParser
> ```
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParserFactory.java` line 24 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncf_6u9AGb5RUTnCp7c:-Ncf_6u9AGb5RUTnCp7d:b-tw0c60) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParserFactory.java#L24)):_
> 
> > ```java
> >   HarnessDataPointParserFactory harnessDataPointParserFactory;
> > 
> >   public DataSourceDataPointParser getDataSourceDataPointParser(String identifier) {
> > ```
> 
> How about renaming it to `getDataPointParserFactory()`?
> 
> _Code quote:_
> 
> ```java
> getDataSourceDataPointParser
> ```
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncfbyr04r337ur6_UUv:-Ncfbyr04r337ur6_UUw:b-86phg1) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java#L21)):_
> 
> > ```java
> > @UtilityClass
> > public class ValueParserUtils {
> >   public Map<String, Object> getMapForPipelineUrlsInfo(String ciPipelineUrl, String cdPipelineUrl) {
> > ```
> 
> Map<String, String> would be fine I guess
> 
> _Code quote:_
> 
> ```java
> Object
> ```
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java` line 31 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-NcfcKEx3uN8g2HFHTs-:-NcfcKEyC_ZwFQplxV_e:b-lm6jmk) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java#L31)):_
> 
> > ```java
> >     Map<String, Object> dataPointsInfo = new HashMap<>();
> >     dataPointsInfo.put(ValueParserConstants.DATA_POINT_VALUE_KEY, dataPointValue);
> >     dataPointsInfo.put(ValueParserConstants.PIPELINE_URLS_FOR_CHECK_FAILURE, urlInfo);
> > ```
> 
> Can we keep this generic? Same keys can be used in Github case as well. How about `{"value": "", "errorMessage": ""}`? cc - @vigneswara-propelo



> > Reviewed 17 of 32 files at r1, 1 of 3 files at r2, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/52057)_ status: 32 file(s) to review. 0 of 2 approvals obtained.
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParser.java` line 9 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncf_MAdEo8u5QTnjZmS:-Ncf_MAdEo8u5QTnjZmT:bqf7nfb) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParser.java#L9)):_
> 
> > ```java
> > package io.harness.idp.scorecard.datapoints.parser;
> > 
> > public interface DataSourceDataPointParser {
> > ```
> 
> How about renaming it to `DataPointParserFactory`?
> 
> _Code quote:_
> 
> ```java
> DataSourceDataPointParser
> ```
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParserFactory.java` line 24 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncf_6u9AGb5RUTnCp7c:-Ncf_6u9AGb5RUTnCp7d:b-tw0c60) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParserFactory.java#L24)):_
> 
> > ```java
> >   HarnessDataPointParserFactory harnessDataPointParserFactory;
> > 
> >   public DataSourceDataPointParser getDataSourceDataPointParser(String identifier) {
> > ```
> 
> How about renaming it to `getDataPointParserFactory()`?
> 
> _Code quote:_
> 
> ```java
> getDataSourceDataPointParser
> ```
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncfbyr04r337ur6_UUv:-Ncfbyr04r337ur6_UUw:b-86phg1) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java#L21)):_
> 
> > ```java
> > @UtilityClass
> > public class ValueParserUtils {
> >   public Map<String, Object> getMapForPipelineUrlsInfo(String ciPipelineUrl, String cdPipelineUrl) {
> > ```
> 
> Map<String, String> would be fine I guess
> 
> _Code quote:_
> 
> ```java
> Object
> ```
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java` line 31 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-NcfcKEx3uN8g2HFHTs-:-NcfcKEyC_ZwFQplxV_e:b-lm6jmk) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java#L31)):_
> 
> > ```java
> >     Map<String, Object> dataPointsInfo = new HashMap<>();
> >     dataPointsInfo.put(ValueParserConstants.DATA_POINT_VALUE_KEY, dataPointValue);
> >     dataPointsInfo.put(ValueParserConstants.PIPELINE_URLS_FOR_CHECK_FAILURE, urlInfo);
> > ```
> 
> Can we keep this generic? Same keys can be used in Github case as well. How about `{"value": "", "errorMessage": ""}`? cc - @vigneswara-propelo
 

all are fixed except How about renaming it to DataPointParserFactory? as this is base class for factory.


> How about renaming it to `DataPointParserFactory`?



> > Reviewed 17 of 32 files at r1, 1 of 3 files at r2, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/52057)_ status: 32 file(s) to review. 0 of 2 approvals obtained.
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParser.java` line 9 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncf_MAdEo8u5QTnjZmS:-Ncf_MAdEo8u5QTnjZmT:bqf7nfb) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParser.java#L9)):_
> 
> > ```java
> > package io.harness.idp.scorecard.datapoints.parser;
> > 
> > public interface DataSourceDataPointParser {
> > ```
> 
> How about renaming it to `DataPointParserFactory`?
> 
> _Code quote:_
> 
> ```java
> DataSourceDataPointParser
> ```
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParserFactory.java` line 24 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncf_6u9AGb5RUTnCp7c:-Ncf_6u9AGb5RUTnCp7d:b-tw0c60) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/DataSourceDataPointParserFactory.java#L24)):_
> 
> > ```java
> >   HarnessDataPointParserFactory harnessDataPointParserFactory;
> > 
> >   public DataSourceDataPointParser getDataSourceDataPointParser(String identifier) {
> > ```
> 
> How about renaming it to `getDataPointParserFactory()`?
> 
> _Code quote:_
> 
> ```java
> getDataSourceDataPointParser
> ```
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-Ncfbyr04r337ur6_UUv:-Ncfbyr04r337ur6_UUw:b-86phg1) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java#L21)):_
> 
> > ```java
> > @UtilityClass
> > public class ValueParserUtils {
> >   public Map<String, Object> getMapForPipelineUrlsInfo(String ciPipelineUrl, String cdPipelineUrl) {
> > ```
> 
> Map<String, String> would be fine I guess
> 
> _Code quote:_
> 
> ```java
> Object
> ```
> 
> _[`idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java` line 31 at r2](https://reviewable.io/reviews/harness/harness-core/52057#-NcfcKEx3uN8g2HFHTs-:-NcfcKEyC_ZwFQplxV_e:b-lm6jmk) ([raw file](https://github.com/harness/harness-core/blob/a929af9db4df37b47a99bbdd54fc669190b3eda2/idp-service/src/main/java/io/harness/idp/scorecard/datapointsdata/datapointvalueparser/ValueParserUtils.java#L31)):_
> 
> > ```java
> >     Map<String, Object> dataPointsInfo = new HashMap<>();
> >     dataPointsInfo.put(ValueParserConstants.DATA_POINT_VALUE_KEY, dataPointValue);
> >     dataPointsInfo.put(ValueParserConstants.PIPELINE_URLS_FOR_CHECK_FAILURE, urlInfo);
> > ```
> 
> Can we keep this generic? Same keys can be used in Github case as well. How about `{"value": "", "errorMessage": ""}`? cc - @vigneswara-propelo



> How about renaming it to `DataPointParserFactory`?

this is not fixed as this is base class for factory so naming it as factory will not be suitable.

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 8 of 32 files at r1, 1 of 3 files at r2, 19 of 19 files at r3, 11 of 11 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52057)* status: 35 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger ut2

**Review by mankrit-singh — DISMISSED** [review]

File: idp-service/src/main/java/io/harness/idp/common/Constants.java [hunk 1]
```diff
@@ -58,4 +58,9 @@ private Constants() {}
   public static final String DOT_SEPARATOR = ".";
   public static final String SPACE_SEPARATOR = " ";
   public static final String SUCCESS_RESPONSE = "SUCCESS";
+
```
Comment (sathish-soundarapandian) [inline idp-service/src/main/java/io/harness/idp/common/Constants.java:61]: This needs to be in sync with what we have defined in the json for DB entities.
Check - 
idp-service/src/main/resources/migrations/scorecard/dataPoints.json
idp-service/src/main/resources/migrations/scorecard/dataSources.json

File: idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/CatalogDataPointParserFactory.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.AllArgsConstructor;
+
+@OwnedBy(HarnessTeam.IDP)
+@AllArgsConstructor(onConstructor = @__({ @Inject }))
+public class CatalogDataPointParserFactory implements DataPointParserFactory {
+  private GenericExpressionParser genericExpressionParser;
+
+  public DataPointParser getParser(String identifier) {
+    switch (identifier) {
+      // Catalog
```
Comment (sathish-soundarapandian) [inline idp-service/src/main/java/io/harness/idp/scorecard/datapoints/parser/CatalogDataPointParserFactory.java:24]: nit: This class will contain parsers only for catalog DPs. So we don't need this explicit comment

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 15 of 15 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52057)* status: 41 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger ut0, ut1

**Comment by devesh-harness** [comment]
trigger review

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 4 of 4 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52057)* status: 41 file(s) to review. 1. unresolved discussions.

**Pull request merged**